src/System.Drawing.Primitives/tests/ColorTests.cs

```diff
@@ -491,5 +493,33 @@ public void DebuggerAttributesAreValid()
             DebuggerAttributes.ValidateDebuggerDisplayReferences(Color.Aquamarine);
             DebuggerAttributes.ValidateDebuggerDisplayReferences(Color.FromArgb(4, 3, 2, 1));
         }
+
+        [Fact]
+        public void UserPreferenceChangingEventTest()
+        {
+            int element = 12; // Win32SystemColors.AppWorkSpace.
+            Color color = System.Drawing.SystemColors.AppWorkspace;
+
+            // A call to ToArgb is necessary before changing the system colors because it initializes the knownColorTable.
+            int oldColor = color.ToArgb();
+
+            try
+            {
+                // choosing a random int instead of a defined color to avoid failures for users with that defined color as system color.
+                Assert.Equal(1, SetSysColors(1, new int[] { element }, new int[] { 89 }));
```

Comment:
Why not set a specific color and then Assert that the value you get is equal? 

[COLORREF](https://docs.microsoft.com/en-us/windows/desktop/gdi/colorref) is defined as 0x00bbggrr, so you just need to encode a color to that integer and then check it.  You should also check that the existing value isn't the color you're changing it to, otherwise your assert will fail.

Also, since SetSysColors is expecting BGR format instead of RGB the call below in the finally does the wrong thing.